### ISSUE TITLE
XMLHttpRequest: readystatechange listener is called twice for unhandled requests

### ISSUE THREAD

**Issue by kettanaito** [mswjs/interceptors#103]
## Code

https://github.com/mswjs/node-request-interceptor/blob/b19c82db06fc9c92a8606bd02ea967a1391d7837/src/interceptors/XMLHttpRequest/XMLHttpRequestOverride.ts#L358-L364

See the workaround test:

https://github.com/mswjs/node-request-interceptor/blob/b19c82db06fc9c92a8606bd02ea967a1391d7837/test/compliance/XMLHttpRequest/events-order.test.ts#L75-L88

## Expected behavior

The order of the XMLHttpRequest events must be this:

```
['loadstart', 1],
['readystatechange', 2],
['readystatechange', 4],
['load', 4],
['loadend', 4],
```

There must be no duplicate `['readystatechange', 4]` event.

## Why this is happening?

XMLHttpRequest polyfill from JSDOM executes the attached `readystatechange` listeners as a part of XMLHttpRequest life-cycle, _which is correct_. However, somehow the way that the listener is attached makes it being called twice.

**Comment by kettanaito** [comment]
The order got reworked in #292 but, alas, the double readystatechange event still happens.

**Comment by kettanaito** [comment]
This has to be solvable now since we've added the logic to prevent nested interceptors from handling the same request in #292. I need to look into this and see what calls readystatechange to `DONE` twice.

**Comment by kettanaito** [comment]
## Released: v0.20.0 🎉

This has been released in v0.20.0!

- 📄 [**Release notes**](https://github.com/mswjs/interceptors/releases/tag/v0.20.0)
- 📦 [npm package](https://www.npmjs.com/package/@mswjs/interceptors/v/0.20.0)

Make sure to always update to the latest version (`npm i @mswjs/interceptors@latest`) to get the newest features and bug fixes.

---

_Predictable release automation by [@ossjs/release](https://github.com/ossjs/release)_.
